code/code/misc/combat.cc

```diff
@@ -2451,7 +2453,12 @@ int TBeing::attackRound(const TBeing * target) const
       bonus += amt;
     }
   }
-  
+
+  // Advanced Offense
+  // For Monks and Thieves. This will be about the same as chivalry
+  if (doesKnowSkill(SKILL_ADVANCED_OFFENSE))
+    bonus += (int)((getSkillValue(SKILL_ADVANCED_OFFENSE) / 4.0) * 3.0);
```

Comment:
You don't need to cast it to int, the cast is implicit because bonus is already an int.